iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -155,11 +156,21 @@ inline uint64_t vector<T, Capacity>::capacity() const
 template <typename T, uint64_t Capacity>
 inline void vector<T, Capacity>::clear()
 {
-    for (uint64_t i = 0u; i < m_size; ++i)
+    if (m_size == 0U)
+    {
+        return;
+    }
+
+    for (uint64_t i = m_size - 1U;; --i)
```

Comment:
This is a pessimization since we now have two checks for termination...
How about ```uint64_t i = m_size; do { at(--i).~T(); } while( i !=0); ```